### ISSUE TITLE
Limited data available on rewards and goals

### ISSUE THREAD

**Issue by pdscti** [Patreon/patreon-php#23]
The API will only return a numeric ID for rewards and goals, making it impossible to obtain information like the reward tier amount or even its name. It would be good if the API could be widened to allow this information to be obtained.

**Comment by 21echoes** [comment]
As per [the JSON:API spec](http://jsonapi.org/), all related data is under the `included` key. You can use the numeric ID to find the relevant data in that `included` array. If you want example code for how to do that, it is provided in [this project's README](https://github.com/Patreon/patreon-php/blob/master/README.md), specifically:

```$patron_response = $api_client->fetch_user();
$patron = $patron_response['data'];
$included = $patron_response['included'];
$pledge = null;
if ($included != null) {
  foreach ($included as $obj) {
    if ($obj["type"] == "pledge" && $obj["relationships"]["creator"]["data"]["id"] == $creator_id) {
      $pledge = $obj;
      break;
    }
  }
}```

As you want reward and goal data, you would use something like: ```$reward_id = null; // Wherever you're currently getting this from
$included = $response['included'];
$reward = null;
if ($included != null) {
  foreach ($included as $obj) {
    if ($obj["type"] == "reward" && $obj["id"] == $reward_id) {
      $reward = $obj;
      break;
    }
  }
}```

**Comment by pdscti** [comment]
Hi

What I'm doing is trying to get a list of rewards I have available for people to pledge for via the API. Using the fetch_campaign_and_patrons() method with my own user account I can get a list of reward IDs in the relationships array of the return. But using the fetch_user() end point the rewards I'm offering aren't in the relationships of the returned account data.

As far as I can tell, the only data I can get is the id of the rewards, which isn't very useful. I think you're saying that there should be more information if a pull down the user data of any patrons I have, but that won't give me a list of all the rewards available, just those any individuals might have signed up for.

Is that right, or am I misunderstanding?

**Comment by 21echoes** [comment]
Sorry I was not clear enough: I am not saying that you need to pull down any of your patrons to see your rewards. What I'm saying is that it is never the case that "the only data I can get is the id of the rewards". The way the [JSON:API spec](http://jsonapi.org/) works is: there are two sections of every api response: `data` and `included`. When you see "only the id of the rewards", that means you're only looking in `data` (specifically, it sounds like you're looking at `response.data.relationships.rewards`). If you go look in `included`, you will see all of the full rewards objects with all of their attributes.

The code snippet I provided in my previous response (now using just the `fetch_campaign` API if that's what you're using):

```
$campaign_response = $api_client->fetch_campaign();
$campaign = $campaign_response['data'];
$included = $campaign_response['included'];
$rewards = null;
if ($included != null) {
  foreach ($included as $obj) {
    if ($obj["type"] == "reward") {
      array_push($rewards, $obj);
    }
  }
}
print_r($rewards);
```

Should provide you a `$rewards` array with all of your rewards.

**Comment by 21echoes** [comment]
Of course! Sorry our documentation wasn't clear enough -- we'll be investing in better documentation & interactive API playgrounds & etc in the coming months. Feel free to write in any time if you need more help!

**Comment by JohnSlaughter** [comment]
I know this is closed, but it was related so I didn't want to open a nw ticket to discuss it.

What about reward history?   I do I find out whether a user (who is currently pledged) was pledged a month ago?  Two months ago?

What if they Pledged for July, but canceled for August, and then renewed for September?  If I'm trying to automate who has access to what, I need to know that info.  Just because they are pledged now doesn't mean they were then.

I know I can download the CSV files and tie things up in a database table, but I was hoping this API would offer some of that in leu of downloading 3 years worth of CSV files.
